wagtail/admin/views/generic/models.py

```diff
@@ -256,17 +257,35 @@ def get_buttons(column, instance, *args, **kwargs):
         )
 
     def _get_custom_column(self, field_name, column_class=Column, **kwargs):
-        label, attr = label_for_field(field_name, self.model, return_attr=True)
+        *relation, field = (
+            [field_name]
+            if hasattr(self.model, field_name)
+            else field_name.split(LOOKUP_SEP)
+        )
+        model_class = self.model
+
+        # iterate over the relation list to try to get the last model
+        # where the field exists
+        foreign_field_name = ""
+        for model in relation:
+            foreign_field = model_class._meta.get_field(model)
+            foreign_field_name += foreign_field.verbose_name + " "
```

Comment:
This can be very long if we traverse several relations, maybe just take the `verbose_name` of the last relation?

```suggestion
            foreign_field_name = foreign_field.verbose_name + " "
```

Reply:
Yeah, I think taking the last relations name only is enough. I wrote like this in the beginning, because in Django Admin it displays the complete relation